### DISCUSSION TITLE
Segmentation fault when running the main version wrf-hydro with crocus

### DISCUSSION THREAD

**Discussion by Lise-Li** [NCAR/wrf_hydro_nwm_public#726] [Q&A]
Hi, I'm learning to use the main version wrf-hydro. I have successfully compiled the wrf-hydro, and I can run the testcase on the website and my domain case without crocus module on. But as soon as I open the crocus module, the model report error 'Segmentation fault'. I have no idea how to solve this problem. Looking forward to your reply.
Best regards,
Lise

**Comment by scrasmussen** [comment]
Hi Lise,
Could you try rebuilding with the debugging mode turned on? With the preferred build method CMake, this would be done with
```
$ mkdir build
$ cd build
$ cmake .. -DHYDRO_D=1
$ make -j 4
```

Which testcase were you building with? Because I don't think any of them have glaciers so I'm not sure they would work. The `namelist.hrldas` file will have to have the CROCUS_nlist option for Crocus to run properly (sorry if this is redundant, just trying to cover all the bases).
```
&CROCUS_nlist
  crocus_opt = 1   ! 0 model is off, 1 model is on
  act_lev = 40     ! 20-40 normal range, 1-50 acceptable
/
```

I'm interested in seeing your results with the debug mode turned on because if WRF-Hydro fails, we would prefer it to fail with a descriptive message rather than a SEGFAULT and would need to fix that. 

Thanks,
Soren

**Reply by Lise-Li** [reply]
Hi @scrasmussen ,
Thank you for your reply! The testcase I use was Croton_NY_testcase without glaciers. And my namelist.hrldas file has the CROCUS option. I'm new in wrf-hydro, would you mind telling me where  the build directory sholud be build? I am creating the build directory in the wrf-hydro/src directory, but I run cmake3 ... DHYDRO_D=1 and it shows the following error.

![error](https://github.com/NCAR/wrf_hydro_nwm_public/assets/145978291/a4ca9cc2-a2d8-4e77-834c-b74626d1a9a5)

Looking forward to your reply!
Best regards,
Lise

**Reply by scrasmussen** [reply]
No problem! And you'll want to put the `build` directory in the top source directory, so it will be `wrf_hydro/build`. Then it should build properly because it needs the setup info from the top `CMakeLists.txt`.

But just to let you know the Croton, NY doesn't have any glacier cells in the input so if you turn on the crocus option you will get the following error:
```
 name = "glacier"
 STOP FATAL ERROR: In module_hrldas_netcdf_io.F get_landuse_or_glacinfo_netcdf() - MODULE_NOAHLSM_HRLDAS_INPUT: nf90_inq_varid for variable glacier
```

You'll need different domain files with glacier cells for the Crocus option to work.

**Reply by Lise-Li** [reply]
Hi @scrasmussen ,

Thank you for your reply! I tried to follow your method to create build directory in the wrf-hydro driectory, but after I run cmake3 ... DHYDRO_D=1, it still show the following errors. The compiler I load was gcc9.3, openmpi/4.1.0. But it seemed that the model used compiler was the computer system default. How can I solve the problem?
[CMakeError.log](https://github.com/NCAR/wrf_hydro_nwm_public/files/13712980/CMakeError.log)
[CMakeOutput.log](https://github.com/NCAR/wrf_hydro_nwm_public/files/13712981/CMakeOutput.log)

Looking forward to your reply.
Best regards,
Lise

**Reply by scrasmussen** [reply]
Hi Lise,

Gfortran 9.3 shouldn't be having a problem with arrays with more than 9 dimensions, so like you said it must be finding the wrong gfortran. The easiest way to fix this would be run CMake while setting the environment variable `FC`.
```bash
$ FC=/path/to/gfortran-9.3 cmake .. -DHYDRO_D=1
```

You could also use `$ export FC=/path/to/gfortran-9.3` and then the command `$ cmake .. -DHYDRO_D=1` if you prefer. Note, you might also need to set the C compiler `CC` as well.
